_search-plugins/knn/approximate-knn.md

```diff
@@ -299,7 +299,7 @@ A space corresponds to the function used to measure the distance between two poi
     <td>\[ d(\mathbf{x}, \mathbf{y}) = 1 - cos { \theta } = 1 - {\mathbf{x} &middot; \mathbf{y} \over \|\mathbf{x}\| &middot; \|\mathbf{y}\|}\]\[ = 1 - 
     {\sum_{i=1}^n x_i y_i \over \sqrt{\sum_{i=1}^n x_i^2} &middot; \sqrt{\sum_{i=1}^n y_i^2}}\]
     where \(\|\mathbf{x}\|\) and \(\|\mathbf{y}\|\) represent the norms of vectors x and y respectively.</td>
-    <td><b>nmslib</b> and <b>faiss:</b>\[ score = {1 \over 1 + d } \]<br><b>Lucene:</b>\[ score = {1 + d \over 2}\]</td>
+    <td><b>nmslib</b> and <b>faiss:</b>\[ score = {1 \over 1 + d } \]<br><b>Lucene:</b>\[ score = {2 - d \over 2}\]</td>
```

Comment:
I'm not sure this is correct formula, as per lucene source code they are using (1 + d) / 2 https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/index/VectorSimilarityFunction.java#L69. 
@SeyedAlirezaFatemi Do you have reference to the formula you're suggesting?

Reply:
In OpenSearch documentation, the distance functions are written and in this case, `d` is the cosine distance function, not the cosine similarity. `(1 + d) / 2` is correct if we assume `d` is the cosine similarity. For `cosinesimil`, `d = 1 - cos(v1, v2)`, so the score function becomes `(2 - d)/2`.

Reply:
I see what you mean, for score function your change is correct, thank you!